lib/pbench/test/unit/agent/test_utils.py

```diff
@@ -0,0 +1,210 @@
+"""Tests for the utils module.
+"""
+
+import os
+import pathlib
+import pytest
+import signal
+import time
+
+from pbench.agent.utils import BaseServer, BaseReturnCode
+
+
+class OurServer(BaseServer):
+    def_port = 4242
+    bad_port_ret_code = 42
+    bad_host_ret_code = 43
+    name = "forty-two"
+
+
+class TestBaseServer:
+    """Verify the utils BaseServer class.
+    """
+
+    def test_constructor(self):
+        with pytest.raises(AssertionError):
+            bs = OurServer("", "")
+
+        for arg1, arg2 in [("", "localhost"), ("localhost", "localhost")]:
+            bs = OurServer(arg1, arg2)
+            assert bs.pid_file is None
+            assert bs.host == "localhost"
+            assert bs.port == 4242
```

Comment:
Shouldn't the RHS be `OurServer.def_port` rather than a copy of the magic number?

Likewise, shouldn't the RHS in line 41 be `OurServer.bad_host_ret_code`?

Etc.

Reply:
Or, if you think accessing the class attributes is too "risky", then it seems like you should be using symbolic constants for them defined earlier and used to initialize the class.

Reply:
Fixed as a separate commit.